src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioVenusSpanMappingService.cs

```diff
@@ -59,6 +65,32 @@ public VisualStudioVenusSpanMappingService(VisualStudioWorkspaceImpl workspace)
             }
         }
 
+        private TextLineCollection GetTextLines(DocumentId currentDocumentId, Location location)
+        {
+            if (location.SourceTree != null)
+            {
+                // normal case - all C# and VB should hit this
+                return location.SourceTree.GetText().Lines;
+            }
+
+            var filePath = location.GetLineSpan().Path;
+            if (filePath != null)
+            {
+                // special case for typescript and etc that don't use our compilations.
+                var documentIds = _workspace.CurrentSolution.GetDocumentIdsWithFilePath(filePath);
+
+                // make sure we are dealing with right document
+                if (documentIds.Contains(currentDocumentId))
```

Comment:
if it was not this late, I would have just put Contract and made it to throw and crash VS, by the way.